src/makim/scheduler.py

```diff
@@ -325,15 +390,32 @@ def remove_job(self, name: str) -> None:
         name : str
             The name of the job to remove.
         """
-        if not self.scheduler:
-            raise RuntimeError('Scheduler not initialized')
+        if self.scheduler is None:
+            MakimLogs.raise_error(
+                'Cannot remove job: Scheduler system is not available',
+                MakimError.SCHEDULER_JOB_ERROR,
+            )
+            return
 
         try:
+            job = self.get_job(name)
+            if not job:
+                MakimLogs.raise_error(
```

Comment:
this `raise_error` is inside the `try` block, but probably it should be kept outside the `try` block.